src/ObjectUtils/DwarfUtils.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ObjectUtils/DwarfUtils.h"
+
+#include <absl/strings/str_cat.h>
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_object_utils {
+
+namespace {
+// This function converts a type `DWARFDie` node into a string representation. Currently, it only
+// handles types that may occur as a type attribute of a DW_TAG_formal_parameter and NOT all
+// possible type Dies from the specification (https://dwarfstd.org/doc/DWARF5.pdf).
+std::string DwarfTypeDieToString(const llvm::DWARFDie& type_die) {
+  ORBIT_CHECK(type_die.isValid());
+
+  // Some Die nodes contain a name directly (e.g. base types or typedefs). We can use that name.
+  if (const char* name = type_die.getName(llvm::DINameKind::LinkageName); name != nullptr) {
+    return {name};
+  }
+
+  std::string result{};
+  const llvm::dwarf::Tag tag = type_die.getTag();
+
+  // Add the type modifiers that occur as prefix, such as "const" or "volatile". We need to identify
+  // those by their tag.
+  // Note that we only handle the modifiers that can occur in C/C++ code here (for further details
+  // see section 5.3 in the DWARF spec: https://dwarfstd.org/doc/DWARF5.pdf).
+  switch (tag) {
+    case llvm::dwarf::DW_TAG_atomic_type:
+      absl::StrAppend(&result, "atomic");
```

Comment:
If we stick with adding parameters also for C-functions, then this should probbably be called `_Atomic` (see https://en.cppreference.com/w/c/language/atomic).